src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/WorkerTest.java

```diff
@@ -429,12 +432,13 @@ public final void testCleanupShardConsumers() {
     }
 
     @Test
-    public final void testInitializationFailureWithRetries() {
+    public final void testInitializationFailureWithRetries() throws Exception {
         String stageName = "testInitializationWorker";
         IRecordProcessorFactory recordProcessorFactory = new TestStreamletFactory(null, null);
         config = new KinesisClientLibConfiguration(stageName, null, null, null);
         int count = 0;
         when(proxy.getShardList()).thenThrow(new RuntimeException(Integer.toString(count++)));
```

Comment:
We shouldn't need this I believe, we will always be using getShards with filter in the empty case. (L450)

Reply:
Will remove this line